test/test_unary_ufuncs.py

```diff
@@ -270,9 +274,27 @@ def test_reference_numerics(self, device, dtype, op):
                                            include_extremal_values=include_extremals)
         for t in tensors:
             if dtype is torch.bfloat16:
-                a = t.cpu().to(torch.float32).numpy()
+                a = t.cpu().to(default_dtype).numpy()
             else:
-                a = t.cpu().numpy()
+                # NumPy promotes integer types to double while
```

Comment:
Let's not change test_unary_ufuncs.py to handle this. This logic should live exclusively in the reference function itself.